web/frontend/src/components/utils/usePostCall.tsx

```diff
@@ -18,7 +18,7 @@ const usePostCall = (setError) => {
     }
```

Comment:
while we're at it, the `setIsPosting` should be in a `finally` block and removed elsewhere:

```ts
catch (error) {
  ...
} finally {
  setIsPosting((prev) => !prev);
}

Reply:
The problem here is that we return a boolean after each setIsPosting(in either try or catch) , should I create a variable before the try and change it in the try/tach then return it in the finally ?

Reply:
Yes, you're right. I would do something like that:

```ts
  return async (endpoint, request, setIsPosting) => {
    let success = true;

    try {
      const response = await fetch(endpoint, request);

      if (!response.ok) {
        const txt = await response.text();
        throw new Error(txt);
      }

      setError(null);
    } catch (error) {
      setError(error.message);
      success = false;
    }

    setIsPosting((prev) => !prev);

    return success;
  };
```

Reply:
I pushed the update